### PR TITLE
[GEOS-9321] WFS GeoJSONBuilder limits max nested level to 20

### PR DESCRIPTION
**Pull request by fernandor777**
This PR upgrades json-lib version and adds testing or the max nested limit issue on WFS GeoJson encoding.

https://osgeo-org.atlassian.net/browse/GEOS-9321

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [x] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

File: src/community/security/keycloak/pom.xml [hunk 1]
```diff
@@ -48,7 +48,6 @@
 		<dependency>
 			<groupId>net.sf.json-lib</groupId>
 			<artifactId>json-lib</artifactId>
-			<classifier>jdk15</classifier>
```
Comment (nmco) [inline src/community/security/keycloak/pom.xml]: Is this related with this fix?

File: src/community/security/oauth2-geonode/pom.xml [hunk 2]
```diff
@@ -47,7 +47,6 @@
 		<dependency>
 			<groupId>net.sf.json-lib</groupId>
 			<artifactId>json-lib</artifactId>
-			<classifier>jdk15</classifier>
```
Comment (nmco) [inline src/community/security/oauth2-geonode/pom.xml]: Same as above.

File: src/community/security/oauth2-github/pom.xml [hunk 3]
```diff
@@ -47,7 +47,6 @@
 		<dependency>
 			<groupId>net.sf.json-lib</groupId>
 			<artifactId>json-lib</artifactId>
-			<classifier>jdk15</classifier>
```
Comment (nmco) [inline src/community/security/oauth2-github/pom.xml]: Same as above.

File: src/community/security/oauth2-openid-connect/pom.xml [hunk 4]
```diff
@@ -50,7 +50,6 @@
 		<dependency>
 			<groupId>net.sf.json-lib</groupId>
 			<artifactId>json-lib</artifactId>
-			<classifier>jdk15</classifier>
```
Comment (nmco) [inline src/community/security/oauth2-openid-connect/pom.xml]: Same as above.

File: src/main/pom.xml [hunk 5]
```diff
@@ -44,7 +44,6 @@
   <dependency>
    <groupId>net.sf.json-lib</groupId>
    <artifactId>json-lib</artifactId>
-   <classifier>jdk15</classifier>
```
Comment (nmco) [inline src/main/pom.xml]: Same as above.

File: src/wfs/pom.xml [hunk 6]
```diff
@@ -60,7 +60,6 @@
     <dependency>
       <groupId>net.sf.json-lib</groupId>
       <artifactId>json-lib</artifactId>
-      <classifier>jdk15</classifier>
```
Comment (nmco) [inline src/wfs/pom.xml]: Same as above.

File: src/community/security/keycloak/pom.xml (hunk 1 above)
Comment (fernandor777) [reply src/community/security/keycloak/pom.xml]: Indeed, it's related to json-lib fork new release version.
